hi @cristhiank , there seems to be a problem. I see the scanner picked the script up:

```
2019-04-02 16:34:40,126 DEBUG [io.qua.fly.run.gra.QuarkusPathLocationScanner] (XNIO-1 task-1) Loading db/migration/V1.0.0__Quarkus.sql
```

but then it fails to actually load these:

```
Caused by: org.flywaydb.core.api.FlywayException: Unable to obtain inputstream for resource: db/migration/V1.0.0__Quarkus.sql
	at org.flywaydb.core.internal.resource.classpath.ClassPathResource.loadAsString(ClassPathResource.java:102)
	at org.flywaydb.core.internal.sqlscript.SqlScript.<init>(SqlScript.java:85)
	at org.flywaydb.core.internal.resolver.sql.SqlMigrationResolver.addMigrations(SqlMigrationResolver.java:140)
```

I can have a look as well but it will have to wait a couple of days

--- a/bom/pom.xml
+++ b/bom/pom.xml
@@ -147,6 +147,7 @@
         <jprocesses.version>1.6.5</jprocesses.version>
         <jboss-logmanager.version>1.0.2</jboss-logmanager.version>
         <jetty.version>9.4.15.v20190215</jetty.version>
+        <flyway.version>5.2.4</flyway.version>
         <yasson.version>1.0.3</yasson.version>
     </properties>
 
@@ -248,6 +249,11 @@
             <dependency>
                 <groupId>io.quarkus</groupId>
                 <artifactId>quarkus-elytron-security</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-flyway</artifactId>
                 <version>${project.version}</version>
             </dependency>
             <dependency>
@@ -1695,6 +1701,11 @@
                 <version>${swagger-ui.version}</version>
             </dependency>
 
+            <dependency>
+                <groupId>org.flywaydb</groupId>
+                <artifactId>flyway-core</artifactId>
+                <version>${flyway.version}</version>
+            </dependency>
         </dependencies>
     </dependencyManagement>
 </project>

--- a/build-parent/pom.xml
+++ b/build-parent/pom.xml
@@ -615,6 +615,16 @@
                 <artifactId>quarkus-kubernetes-spi</artifactId>
                 <version>${project.version}</version>
             </dependency>
+            <dependency>
+                <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-flyway</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-flyway-deployment</artifactId>
+                <version>${project.version}</version>
+            </dependency>
 
             <!-- Miscellaneous -->
             <dependency>
@@ -699,7 +709,8 @@
                             </goals>
                             <phase>compile</phase>
                             <configuration>
-                                <deployment>${project.groupId}:${project.artifactId}-deployment:${project.version}</deployment>
+                                <deployment>${project.groupId}:${project.artifactId}-deployment:${project.version}
+                                </deployment>
                             </configuration>
                         </execution>
                     </executions>
@@ -719,8 +730,10 @@
                                     <bannedDependencies>
                                         <excludes>
                                             <!-- Use javax.annotation-api -->
-                                            <exclude>org.jboss.spec.javax.annotation:jboss-annotations-api_1.2_spec</exclude>
-                                            <exclude>org.jboss.spec.javax.annotation:jboss-annotations-api_1.3_spec</exclude>
+                                            <exclude>org.jboss.spec.javax.annotation:jboss-annotations-api_1.2_spec
+                                            </exclude>
+                                            <exclude>org.jboss.spec.javax.annotation:jboss-annotations-api_1.3_spec
+                                            </exclude>
                                             <!-- use our jboss-logmanager -->
                                             <exclude>org.jboss.logging:jboss-logmanager</exclude>
                                             <!-- We don't want all the API's in one jar-->

--- a/core/deployment/src/main/java/io/quarkus/deployment/builditem/FeatureBuildItem.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/builditem/FeatureBuildItem.java
@@ -47,6 +47,7 @@ public final class FeatureBuildItem extends MultiBuildItem {
     public static final String UNDERTOW_WEBSOCKETS = "undertow-websockets";
     public static final String VERTX = "vertx";
     public static final String VERTX_WEB = "vertx-web";
+    public static final String FLYWAY = "flyway";
 
     private final String info;
 

--- a/devtools/common/src/main/filtered/extensions.json
+++ b/devtools/common/src/main/filtered/extensions.json
@@ -393,5 +393,15 @@
     ],
     "groupId": "io.quarkus",
     "artifactId": "quarkus-vertx"
+  },
+  {
+    "name": "Flyway",
+    "labels": [
+      "flyway",
+      "database",
+      "data"
+    ],
+    "groupId": "io.quarkus",
+    "artifactId": "quarkus-flyway"
   }
 ]

--- a/docs/src/main/asciidoc/flyway-guide.adoc
+++ b/docs/src/main/asciidoc/flyway-guide.adoc
@@ -1,0 +1,103 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+
+include::./attributes.adoc[]
+= {project-name} - Using Flyway
+:migrations-path: src/main/resources/db/migration
+:config-file: application.properties
+
+https://flywaydb.org/[Flyway] is a popular database migration tool that is commonly used in JVM environments.
+
+{project-name} provides first class support for using Flyway in native mode as will be explained in this guide.
+
+[NOTE]
+--
+You only need to add this extension if your are doing native compilation with {project-name}.
+Otherwise, you can use Flyway by just adding it as a dependency in the `pom.xml` and run the {project-name} app in JVM mode.
+--
+
+== Setting up native compilation support for Flyway
+
+To get native compilation working in {project-name} with Flyway, you just need to:
+
+* add your migrations to the `{migrations-path}` folder as you usually do with Flyway
+
+NOTE: The migrations location is not configurable in native mode. The Flyway extension will always look for migrations in the default path.
+
+* create your `Flyway` object and run your migration as you normally do
+
+In your `pom.xml`, add the following dependencies:
+
+* the Flyway extension
+* your JDBC driver extension (`quarkus-jdbc-postgresql`, `quarkus-jdbc-h2`, `quarkus-jdbc-mariadb`, ...)
+
+[source,xml]
+--
+<dependencies>
+    <!-- Flyway specific dependencies -->
+    <dependency>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-flyway</artifactId>
+    </dependency>
+
+    <!-- JDBC driver dependencies -->
+    <dependency>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-jdbc-postgresql</artifactId>
+    </dependency>
+</dependencies>
+--
+
+You can use MP-config to get any relevant configuration properties you may need by adding them to the `{config-file}` file.
+
+[source,properties]
+--
+# configure your datasource
+datasource.url: jdbc:postgresql://localhost:5432/mydatabase
+datasource.username: sarah
+datasource.password: connor
+--
+
+Note that you can use any name for these configuration properties.
+
+Add a SQL migration to the default folder following the Flyway naming conventions: `{migrations-path}/V1.0.0__Quarkus.sql`
+
+[source,sql]
+--
+CREATE TABLE quarkus
+(
+  id   INT,
+  name VARCHAR(20)
+);
+INSERT INTO quarkus(id, name)
+VALUES (1, 'QUARKED');
+--
+
+Now you can create your `Flyway` object as you usually do:
+
+[source,java]
+--
+@ApplicationScoped
+public class MigrationService {
+
+    @ConfigProperty(name = "datasource.url") <1>
+    String dbURL;
+    @ConfigProperty(name = "datasource.username")
+    String dbUser;
+    @ConfigProperty(name = "datasource.password")
+    String dbPassword;
+
+    public void migrate() {
+        Flyway flyway = Flyway.configure()
+                        .dataSource(dbURL, dbUser, dbPassword)
+                        .load(); <2>
+        flyway.migrate();
+    }
+}
+--
+
+<1> Inject your config properties as needed
+<2> Create the `Flyway` object and enjoy

--- a/extensions/flyway/deployment/pom.xml
+++ b/extensions/flyway/deployment/pom.xml
@@ -1,0 +1,58 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2019 Red Hat, Inc.
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>quarkus-flyway-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-flyway-deployment</artifactId>
+    <name>Quarkus - Flyway - Deployment</name>
+
+    <dependencies>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-core-deployment</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-flyway</artifactId>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <annotationProcessorPaths>
+                        <path>
+                            <groupId>io.quarkus</groupId>
+                            <artifactId>quarkus-extension-processor</artifactId>
+                            <version>${project.version}</version>
+                        </path>
+                    </annotationProcessorPaths>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+</project>

--- a/extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayBuildStep.java
+++ b/extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayBuildStep.java
@@ -1,0 +1,109 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.quarkus.flyway;
+
+import static java.nio.file.Files.walk;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.ApplicationArchivesBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.substrate.SubstrateResourceBuildItem;
+
+class FlywayBuildStep {
+
+    private static final String FLYWAY_DATABASES_PATH_ROOT = "org/flywaydb/core/internal/database";
+    private static final String FLYWAY_METADATA_TABLE_FILENAME = "createMetaDataTable.sql";
+    private static final String[] FLYWAY_DATABASES_WITH_SQL_FILE = {
+            "cockroachdb",
+            "derby",
+            "h2",
+            "hsqldb",
+            "mysql",
+            "oracle",
+            "postgresql",
+            "redshift",
+            "saphana",
+            "sqlite",
+            "sybasease"
+    };
+
+    private static final Logger LOGGER = Logger.getLogger(FlywayBuildStep.class);
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FeatureBuildItem.FLYWAY);
+    }
+
+    @Record(ExecutionTime.STATIC_INIT)
+    @BuildStep(providesCapabilities = "io.quarkus.flyway")
+    void registerSubstrateResources(
+            BuildProducer<SubstrateResourceBuildItem> resource,
+            ApplicationArchivesBuildItem appArchives) throws IOException, URISyntaxException {
+        Path root = appArchives.getRootArchive().getArchiveRoot();
+        List<String> resources = generateDatabasesSQLFiles();
+        resources.addAll(discoverApplicationMigrations(root));
+        resource.produce(new SubstrateResourceBuildItem(resources.toArray(new String[0])));
+    }
+
+    private List<String> discoverApplicationMigrations(Path root) throws IOException, URISyntaxException {
+        List<String> resources = new ArrayList<>();
+        try {
+            // TODO: this should be configurable, using flyway default
+            String location = "db/migration";
+            Enumeration<URL> migrations = Thread.currentThread().getContextClassLoader().getResources(location);
+            while (migrations.hasMoreElements()) {
+                URL path = migrations.nextElement();
+                LOGGER.info("Adding application migrations in path: " + path);
+                List<String> applicationMigrations = walk(Paths.get(path.toURI()))
+                        .filter(Files::isRegularFile)
+                        .map(it -> Paths.get(location, it.getFileName().toString()).toString())
+                        .peek(it -> LOGGER.debug("Discovered: " + it))
+                        .collect(Collectors.toList());
+                resources.addAll(applicationMigrations);
+            }
+            return resources;
+        } catch (IOException | URISyntaxException e) {
+            throw e;
+        }
+    }
+
+    private List<String> generateDatabasesSQLFiles() {
+        List<String> result = new ArrayList<>();
+        for (String database : FLYWAY_DATABASES_WITH_SQL_FILE) {
+            String filePath = FLYWAY_DATABASES_PATH_ROOT + "/" + database + "/" + FLYWAY_METADATA_TABLE_FILENAME;
+            result.add(filePath);
+            LOGGER.debug("Adding flyway internal migration: " + filePath);
+        }
+        return result;
+    }
+}

--- a/extensions/flyway/pom.xml
+++ b/extensions/flyway/pom.xml
@@ -1,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2019 Red Hat, Inc.
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <parent>
+        <artifactId>quarkus-build-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+        <relativePath>../../build-parent/pom.xml</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-flyway-parent</artifactId>
+    <name>Quarkus - Flyway</name>
+    <packaging>pom</packaging>
+    <modules>
+        <module>runtime</module>
+        <module>deployment</module>
+    </modules>
+</project>

--- a/extensions/flyway/runtime/pom.xml
+++ b/extensions/flyway/runtime/pom.xml
@@ -1,0 +1,62 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2019 Red Hat, Inc.
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>quarkus-flyway-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-flyway</artifactId>
+    <name>Quarkus - Flyway - Runtime</name>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.flywaydb</groupId>
+            <artifactId>flyway-core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.oracle.substratevm</groupId>
+            <artifactId>svm</artifactId>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>io.quarkus.bootstrap</groupId>
+                <artifactId>quarkus-bootstrap-maven-plugin</artifactId>
+            </plugin>
+            <plugin>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <annotationProcessorPaths>
+                        <path>
+                            <groupId>io.quarkus</groupId>
+                            <artifactId>quarkus-extension-processor</artifactId>
+                            <version>${project.version}</version>
+                        </path>
+                    </annotationProcessorPaths>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+</project>

--- a/extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/graal/CompositeMigrationResolverSubstitutions.java
+++ b/extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/graal/CompositeMigrationResolverSubstitutions.java
@@ -1,0 +1,70 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.quarkus.flyway.runtime.graal;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.flywaydb.core.api.configuration.Configuration;
+import org.flywaydb.core.api.resolver.MigrationResolver;
+import org.flywaydb.core.internal.clazz.ClassProvider;
+import org.flywaydb.core.internal.database.base.Database;
+import org.flywaydb.core.internal.resolver.java.JavaMigrationResolver;
+import org.flywaydb.core.internal.resolver.jdbc.JdbcMigrationResolver;
+import org.flywaydb.core.internal.resolver.sql.SqlMigrationResolver;
+import org.flywaydb.core.internal.resource.ResourceProvider;
+import org.flywaydb.core.internal.sqlscript.SqlStatementBuilderFactory;
+
+import com.oracle.svm.core.annotate.Alias;
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+@TargetClass(className = "org.flywaydb.core.internal.resolver.CompositeMigrationResolver")
+public final class CompositeMigrationResolverSubstitutions {
+    @Alias
+    private Collection<MigrationResolver> migrationResolvers = new ArrayList<>();
+
+    /**
+     * Substitution to remove Spring-data migration resolver as the spring-data dependency is optional.
+     * This method removes the inclusion of {@link org.flywaydb.core.internal.resolver.spring.SpringJdbcMigrationResolver}
+     * in the resolvers list to avoid native image errors because of the incomplete classpath
+     *
+     * @see org.flywaydb.core.internal.resolver.spring.SpringJdbcMigrationResolver
+     * @see org.flywaydb.core.internal.resolver.CompositeMigrationResolver#CompositeMigrationResolver(Database,
+     *      ResourceProvider, ClassProvider, Configuration, SqlStatementBuilderFactory, MigrationResolver...)
+     */
+    @Substitute
+    public CompositeMigrationResolverSubstitutions(
+            Database database,
+            ResourceProvider resourceProvider,
+            ClassProvider classProvider,
+            Configuration configuration,
+            SqlStatementBuilderFactory sqlStatementBuilderFactory,
+            MigrationResolver... customMigrationResolvers) {
+        if (!configuration.isSkipDefaultResolvers()) {
+            migrationResolvers.add(new SqlMigrationResolver(
+                    database,
+                    resourceProvider,
+                    sqlStatementBuilderFactory,
+                    configuration));
+            migrationResolvers.add(new JavaMigrationResolver(classProvider, configuration));
+            migrationResolvers.add(new JdbcMigrationResolver(classProvider, configuration));
+        }
+        migrationResolvers.addAll(Arrays.asList(customMigrationResolvers));
+    }
+}

--- a/extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/graal/FeatureDetectorSubstitutions.java
+++ b/extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/graal/FeatureDetectorSubstitutions.java
@@ -1,0 +1,66 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.quarkus.flyway.runtime.graal;
+
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+@Substitute
+@TargetClass(className = "org.flywaydb.core.internal.util.FeatureDetector")
+public final class FeatureDetectorSubstitutions {
+
+    @Substitute
+    public FeatureDetectorSubstitutions(ClassLoader classLoader) {
+
+    }
+
+    @Substitute
+    public boolean isApacheCommonsLoggingAvailable() {
+        return false;
+    }
+
+    @Substitute
+    public boolean isSlf4jAvailable() {
+        return false;
+    }
+
+    @Substitute
+    public boolean isSpringJdbcAvailable() {
+        return false;
+    }
+
+    @Substitute
+    public boolean isJBossVFSv2Available() {
+        return false;
+    }
+
+    @Substitute
+    public boolean isJBossVFSv3Available() {
+        return false;
+    }
+
+    @Substitute
+    public boolean isOsgiFrameworkAvailable() {
+        return false;
+    }
+
+    @Substitute
+    public boolean isAndroidAvailable() {
+        return false;
+    }
+
+}

--- a/extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/graal/PostgreSQLSqlStatementBuilderSubstitutions.java
+++ b/extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/graal/PostgreSQLSqlStatementBuilderSubstitutions.java
@@ -1,0 +1,69 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.quarkus.flyway.runtime.graal;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BooleanSupplier;
+
+import org.flywaydb.core.internal.line.Line;
+import org.flywaydb.core.internal.sqlscript.Delimiter;
+import org.flywaydb.core.internal.sqlscript.SqlStatement;
+import org.flywaydb.core.internal.sqlscript.StandardSqlStatement;
+
+import com.oracle.svm.core.annotate.Alias;
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+/**
+ * This substitution removes de PosgreSQL COPY statement support in Flyway to allow native image compilation
+ * when the PostgreSQL Driver is not in the classpath.
+ */
+@TargetClass(className = "org.flywaydb.core.internal.database.postgresql.PostgreSQLSqlStatementBuilder", onlyWith = PostgreSQLSqlStatementBuilderSubstitutions.Selector.class)
+public final class PostgreSQLSqlStatementBuilderSubstitutions {
+    @Alias
+    protected List<Line> lines = new ArrayList<>();
+    @Alias
+    protected Delimiter delimiter;
+    @Alias
+    private boolean pgCopy;
+
+    /**
+     * Returns only a {@link StandardSqlStatement} as the normal SQL migrations do or throw an exception
+     * if pgCopy flag is true
+     */
+    @Substitute
+    public SqlStatement getSqlStatement() {
+        if (pgCopy) {
+            throw new IllegalStateException("pgCopy is not supported yet!");
+        }
+        return new StandardSqlStatement(lines, delimiter);
+    }
+
+    static final class Selector implements BooleanSupplier {
+
+        @Override
+        public boolean getAsBoolean() {
+            try {
+                Class.forName("org.postgresql.Driver");
+                return false;
+            } catch (ClassNotFoundException e) {
+                return true;
+            }
+        }
+    }
+}

--- a/extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/graal/QuarkusPathLocationScanner.java
+++ b/extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/graal/QuarkusPathLocationScanner.java
@@ -1,0 +1,155 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.quarkus.flyway.runtime.graal;
+
+import java.io.IOException;
+import java.net.JarURLConnection;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+import org.flywaydb.core.api.Location;
+import org.flywaydb.core.api.logging.Log;
+import org.flywaydb.core.api.logging.LogFactory;
+import org.flywaydb.core.internal.resource.LoadableResource;
+import org.flywaydb.core.internal.resource.classpath.ClassPathResource;
+import org.flywaydb.core.internal.scanner.classpath.ResourceAndClassScanner;
+
+public final class QuarkusPathLocationScanner implements ResourceAndClassScanner {
+    private static final Log LOG = LogFactory.getLog(QuarkusPathLocationScanner.class);
+    // TODO: this should be configurable, using flyway default
+    private final static String DEFAULT_NATIVE_LOCATION = "db/migration";
+    private static Set<String> ALL_SQL_RESOURCES;
+
+    static {
+        try {
+            ALL_SQL_RESOURCES = discoverApplicationMigrations();
+        } catch (IOException | URISyntaxException e) {
+            throw new IllegalStateException("Could not discover Flyway migrations", e);
+        }
+    }
+
+    public QuarkusPathLocationScanner(Location location) {
+        if (!location.getPath().equals(DEFAULT_NATIVE_LOCATION)) {
+            LOG.error("Invalid migrations location: " + location.getPath() + ". Flyway migrations must be located in  "
+                    + DEFAULT_NATIVE_LOCATION);
+        }
+    }
+
+    private static Set<String> discoverApplicationMigrations() throws IOException, URISyntaxException {
+        Set<String> resources = new HashSet<>();
+        try {
+            Enumeration<URL> migrations = Thread.currentThread().getContextClassLoader().getResources(
+                    DEFAULT_NATIVE_LOCATION);
+            while (migrations.hasMoreElements()) {
+                URL path = migrations.nextElement();
+                LOG.debug("Adding application migrations in path: " + path);
+                if ("jar".equals(path.getProtocol())) {
+                    resources = scanInJar(path);
+                } else {
+                    throw new IllegalStateException("Expecting jar file. Protocol not supported:" + path.getProtocol());
+                }
+            }
+            return resources;
+        } catch (IOException e) {
+            throw e;
+        }
+    }
+
+    public static Set<String> scanInJar(URL locationUrl) {
+        JarFile jarFile;
+        try {
+            jarFile = getJarFromUrl(locationUrl);
+        } catch (IOException e) {
+            LOG.warn("Unable to determine jar from url (" + locationUrl + "): " + e.getMessage());
+            return Collections.emptySet();
+        }
+
+        try {
+            return findResourceNamesFromJarFile(jarFile);
+        } finally {
+            try {
+                jarFile.close();
+            } catch (IOException e) {
+                // Ignore
+            }
+        }
+    }
+
+    private static JarFile getJarFromUrl(URL locationUrl) throws IOException {
+        URLConnection con = locationUrl.openConnection();
+        if (con instanceof JarURLConnection) {
+            // Should usually be the case for traditional JAR files.
+            JarURLConnection jarCon = (JarURLConnection) con;
+            jarCon.setUseCaches(false);
+            return jarCon.getJarFile();
+        }
+        throw new IllegalStateException("Could not open the jar file " + locationUrl);
+    }
+
+    private static Set<String> findResourceNamesFromJarFile(JarFile jarFile) {
+        Set<String> resourceNames = new TreeSet<>();
+        Enumeration<JarEntry> entries = jarFile.entries();
+        while (entries.hasMoreElements()) {
+            String entryName = entries.nextElement().getName();
+            if (entryName.startsWith(DEFAULT_NATIVE_LOCATION) && !entryName.endsWith("/")) {
+                LOG.debug("Discovered " + entryName);
+                resourceNames.add(entryName);
+            }
+        }
+
+        return resourceNames;
+    }
+
+    /**
+     * Scans the classpath for resources under the configured DEFAULT_NATIVE_LOCATION.
+     *
+     * @return The resources that were found.
+     */
+    @Override
+    public Collection<LoadableResource> scanForResources() {
+        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+        Set<LoadableResource> resources = new HashSet<>();
+        Location defaultLocation = new Location(DEFAULT_NATIVE_LOCATION);
+        for (String file : ALL_SQL_RESOURCES) {
+            LOG.debug("Loading " + file);
+            resources.add(new ClassPathResource(defaultLocation, file, classLoader, StandardCharsets.UTF_8));
+        }
+        return resources;
+    }
+
+    /**
+     * Scans the classpath for concrete classes under the specified package implementing this interface.
+     * Non-instantiable abstract classes are filtered out.
+     *
+     * @return The non-abstract classes that were found.
+     */
+    @Override
+    public Collection<Class<?>> scanForClasses() {
+        // Classes are not supported in native mode
+        return Collections.emptyList();
+    }
+}

--- a/extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/graal/ScannerSubstitutions.java
+++ b/extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/graal/ScannerSubstitutions.java
@@ -1,0 +1,58 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.quarkus.flyway.runtime.graal;
+
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.flywaydb.core.api.Location;
+import org.flywaydb.core.internal.resource.LoadableResource;
+import org.flywaydb.core.internal.scanner.classpath.ResourceAndClassScanner;
+
+import com.oracle.svm.core.annotate.Alias;
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+/**
+ * This substitution replaces the Flyway dynamic scanners with a fixed path scanner in native mode
+ */
+@TargetClass(className = "org.flywaydb.core.internal.scanner.Scanner")
+public final class ScannerSubstitutions {
+
+    @Alias
+    private List<LoadableResource> resources = new ArrayList<>();
+    @Alias
+    private List<Class<?>> classes = new ArrayList<>();
+
+    /**
+     * Creates only {@link QuarkusPathLocationScanner} instances.
+     * Replaces the original method that tries to detect migrations using reflection techniques that are not allowed
+     * in native mode
+     *
+     * @see org.flywaydb.core.internal.scanner.Scanner#Scanner(Collection, ClassLoader, Charset)
+     */
+    @Substitute
+    public ScannerSubstitutions(Collection<Location> locations, ClassLoader classLoader, Charset encoding) {
+        for (Location location : locations) {
+            ResourceAndClassScanner quarkusScanner = new QuarkusPathLocationScanner(location);
+            resources.addAll(quarkusScanner.scanForResources());
+            classes.addAll(quarkusScanner.scanForClasses());
+        }
+    }
+}

--- a/extensions/pom.xml
+++ b/extensions/pom.xml
@@ -101,6 +101,9 @@
 
         <!-- Kubernetes -->
         <module>kubernetes</module>
+
+        <!-- Database migrations -->
+        <module>flyway</module>
     </modules>
 
 </project>

--- a/integration-tests/flyway/pom.xml
+++ b/integration-tests/flyway/pom.xml
@@ -1,0 +1,150 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2019 Red Hat, Inc.
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>quarkus-integration-tests-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+        <relativePath>../</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-integration-test-flyway</artifactId>
+    <name>Quarkus - Integration Tests - Flyway</name>
+    <description>Module that contains Flyway related tests</description>
+    <dependencies>
+        <!-- Enables the Flyway capabilities -->
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-flyway</artifactId>
+        </dependency>
+        <!-- We'll be using the H2 database -->
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-jdbc-h2</artifactId>
+        </dependency>
+        <!-- .. and some REST endpoints -->
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-resteasy</artifactId>
+        </dependency>
+
+        <!-- test dependencies -->
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-junit5</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-test-h2</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.rest-assured</groupId>
+            <artifactId>rest-assured</artifactId>
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.commons</groupId>
+                    <artifactId>commons-lang3</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <resources>
+            <resource>
+                <directory>src/main/resources</directory>
+                <filtering>true</filtering>
+            </resource>
+        </resources>
+        <plugins>
+            <plugin>
+                <groupId>${project.groupId}</groupId>
+                <artifactId>quarkus-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>build</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+
+    <profiles>
+        <profile>
+            <id>native-image</id>
+            <activation>
+                <property>
+                    <name>native</name>
+                </property>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-failsafe-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <goals>
+                                    <goal>integration-test</goal>
+                                    <goal>verify</goal>
+                                </goals>
+                                <configuration>
+                                    <systemProperties>
+                                        <native.image.path>
+                                            ${project.build.directory}/${project.build.finalName}-runner
+                                        </native.image.path>
+                                    </systemProperties>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                    <plugin>
+                        <groupId>${project.groupId}</groupId>
+                        <artifactId>quarkus-maven-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>native-image</id>
+                                <goals>
+                                    <goal>native-image</goal>
+                                </goals>
+                                <configuration>
+                                    <reportErrorsAtRuntime>false</reportErrorsAtRuntime>
+                                    <cleanupServer>true</cleanupServer>
+                                    <enableHttpUrlHandler>true</enableHttpUrlHandler>
+                                    <graalvmHome>${graalvmHome}</graalvmHome>
+                                    <enableJni>false</enableJni>
+                                    <debugBuildProcess>false</debugBuildProcess>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+
+    </profiles>
+
+</project>

--- a/integration-tests/flyway/src/main/java/io/quarkus/example/flyway/FlywayFunctionalityResource.java
+++ b/integration-tests/flyway/src/main/java/io/quarkus/example/flyway/FlywayFunctionalityResource.java
@@ -1,0 +1,70 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.quarkus.example.flyway;
+
+import static java.sql.DriverManager.getConnection;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.flywaydb.core.Flyway;
+
+@Path("/flyway")
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+public class FlywayFunctionalityResource {
+    @ConfigProperty(name = "datasource.url")
+    String dbURL;
+    @ConfigProperty(name = "datasource.username")
+    String dbUser;
+    @ConfigProperty(name = "datasource.password")
+    String dbPassword;
+
+    @GET
+    @Path("/migrate")
+    public String doMigrate() throws SQLException {
+        Flyway flyway = Flyway.configure()
+                .dataSource(dbURL, dbUser, dbPassword)
+                .load();
+        flyway.migrate();
+        int rows = countTableRows();
+        return "OK " + rows;
+    }
+
+    private int countTableRows() throws SQLException {
+        try (Connection connection = getConnection(dbURL, dbUser, dbPassword)) {
+            try (Statement statement = connection.createStatement()) {
+                try (ResultSet execute = statement.executeQuery("SELECT COUNT(*) FROM quarkus; ")) {
+                    if (execute.next()) {
+                        return execute.getInt(1);
+                    } else {
+                        return -1; //ERROR
+                    }
+                }
+            }
+        }
+    }
+}

--- a/integration-tests/flyway/src/main/resources/application.properties
+++ b/integration-tests/flyway/src/main/resources/application.properties
@@ -1,0 +1,21 @@
+#
+# Copyright 2019 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+datasource.url=jdbc:h2:tcp://localhost/mem:test;DB_CLOSE_DELAY=-1
+datasource.username=sa
+datasource.password=sa
+quarkus.log.console.level=DEBUG
+quarkus.log.category."org.flywaydb.core".level=DEBUG
+quarkus.log.category."io.quarkus.flyway".level=DEBUG

--- a/integration-tests/flyway/src/main/resources/db/migration/V1.0.0__Quarkus.sql
+++ b/integration-tests/flyway/src/main/resources/db/migration/V1.0.0__Quarkus.sql
@@ -1,0 +1,7 @@
+CREATE TABLE quarkus
+(
+  id   INT,
+  name VARCHAR(20)
+);
+INSERT INTO quarkus(id, name)
+VALUES (1, 'QUARKED');

--- a/integration-tests/flyway/src/test/java/io/quarkus/example/flyway/FlywayFunctionalityNativeIT.java
+++ b/integration-tests/flyway/src/test/java/io/quarkus/example/flyway/FlywayFunctionalityNativeIT.java
@@ -1,0 +1,23 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.quarkus.example.flyway;
+
+import io.quarkus.test.junit.SubstrateTest;
+
+@SubstrateTest
+public class FlywayFunctionalityNativeIT extends FlywayFunctionalityTest {
+}

--- a/integration-tests/flyway/src/test/java/io/quarkus/example/flyway/FlywayFunctionalityTest.java
+++ b/integration-tests/flyway/src/test/java/io/quarkus/example/flyway/FlywayFunctionalityTest.java
@@ -1,0 +1,36 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.quarkus.example.flyway;
+
+import static io.restassured.RestAssured.when;
+import static org.hamcrest.Matchers.is;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+@DisplayName("Tests flyway extension")
+public class FlywayFunctionalityTest {
+
+    @Test
+    @DisplayName("Migrates a schema correctly")
+    public void testFlywayFunctionality() throws Exception {
+        when().get("/flyway/migrate").then().body(is("OK 1"));
+    }
+}

--- a/integration-tests/flyway/src/test/java/io/quarkus/example/flyway/FlywayTestResources.java
+++ b/integration-tests/flyway/src/test/java/io/quarkus/example/flyway/FlywayTestResources.java
@@ -1,0 +1,24 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.quarkus.example.flyway;
+
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.h2.H2DatabaseTestResource;
+
+@QuarkusTestResource(H2DatabaseTestResource.class)
+public class FlywayTestResources {
+}

--- a/integration-tests/pom.xml
+++ b/integration-tests/pom.xml
@@ -55,6 +55,7 @@
         <module>camel-core</module>
         <module>camel-salesforce</module>
         <module>camel-aws-s3</module>
+        <module>flyway</module>
     </modules>
 
     <build>

